src/frequenz/sdk/actor/_runner.py

```diff
@@ -0,0 +1,38 @@
+# License: MIT
+# Copyright © 2023 Frequenz Energy-as-a-Service GmbH
+
+"""Utility functions to run and synchronize the execution of actors."""
+
+
+import asyncio
+import logging
+from typing import Any
+
+from ._decorator import BaseActor
+
+log = logging.getLogger(__name__)
+
+
+async def run(*actors: Any) -> None:
+    """Await the completion of all actors.
+
+    Args:
+        actors: the actors to be awaited.
+
+    Raises:
+        AssertionError: if any of the actors is not an instance of BaseActor.
+    """
+    actor_tasks = []
+    # Check that each actor is an instance of BaseActor at runtime,
+    # due to the indirection created by the actor decorator.
+    for actor in actors:
+        assert isinstance(actor, BaseActor), f"{actor} is not an instance of BaseActor"
+        actor_tasks.append(asyncio.create_task(actor.join()))
+
+    # Currently the actor decorator manages the life-cycle of the actor tasks
+    while actor_tasks:
+        done, actor_tasks = await asyncio.wait(
+            actor_tasks, return_when=asyncio.FIRST_COMPLETED
+        )
+        log.debug("Actor task done: %s", done)
+        log.debug("Actor task pending: %s", actor_tasks)
```

Comment:
@leandro-lucarella-frequenz I'm happy to remove the logging if you think is not needed.
And what about `actor_tasks`? Should I better rename it to something like `actor_tasks_pending` instead? Pls let me know your thoughts

Reply:
I would just use `done, pending` as it is what we normally use. In any case, I would make done and pending symmetric in terms of name (`done`/`pendding`, `done_tasks`/`pending_tasks`, etc.)